### PR TITLE
support reloading for int8 models

### PR DESCRIPTION
**Pull request by xin3he**
# What does this PR do?

In this PR we enhance the example to support reloading for int8 models. users and reload quantized model from `output_dir` or `model_name` on [model hub](https://huggingface.co/models?other=Intel%C2%AE%20Neural%20Compressor).

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

**Review by PenghuiCheng — APPROVED** [review]

File: examples/text-classification/run_glue.py [hunk 1]
```diff
@@ -571,162 +587,169 @@ def take_train_steps(model, trainer, resume_from_checkpoint, last_checkpoint):
...
+        if optim_args.load_int8:
+            loaded_model = model
+        else:
+            loaded_model = IncQuantizedModelForSequenceClassification.from_pretrained(training_args.output_dir)
         loaded_model.eval()
+
         result_loaded_model = take_eval_steps(loaded_model, trainer, metric_name)
 
-        if result_loaded_model != result_optimized_model:
+        if not optim_args.load_int8 and result_loaded_model != result_optimized_model:
```
Comment (echarlaix) [inline examples/text-classification/run_glue.py:752]: Not sure about the case where `optim_args.load_int8` is set to `True` as we don't have the original model's accuracy to compare it with

**Review by echarlaix — COMMENTED** [review]
Great addition @xin3he, thanks !

File: examples/text-classification/README.md [hunk 2]
```diff
@@ -36,6 +36,17 @@ python run_glue.py \
     --verify_loading \
     --output_dir /tmp/sst2_output
 ```
+The following example reloads the quantized DistilBERT and verifies the accuracy on sst-2 task. [Int8 models](https://huggingface.co/models?other=Intel%C2%AE%20Neural%20Compressor) on model hub are also available.
+
+```bash
+python run_glue.py \
+    --model_name_or_path /tmp/sst2_output \
```
[the remark this answers is not in the document]
Comment (xin3he) [reply examples/text-classification/README.md:43]: Yes, but my concern is keeping the series with the tuning command. I will add one more intro. about loading int8 models from the [Intel](https://huggingface.co/Intel) org

File: examples/text-classification/run_glue.py [hunk 3]
```diff
@@ -571,162 +587,169 @@ def take_train_steps(model, trainer, resume_from_checkpoint, last_checkpoint):
     def train_func(model):
         return take_train_steps(model, trainer, resume_from_checkpoint, last_checkpoint)
 
-    quantizer = None
-    pruner = None
-    distiller = None
+    if optim_args.apply_quantization or optim_args.apply_pruning or optim_args.apply_distillation:
 
-    if not optim_args.apply_quantization and not optim_args.apply_pruning and not optim_args.apply_distillation:
```
[the remark this answers is not in the document]
Comment (xin3he) [reply examples/text-classification/run_glue.py:578]: Good suggestion, thanks

File: examples/text-classification/run_glue.py (hunk 1 above)
Comment (xin3he) [reply examples/text-classification/run_glue.py:752]: Yes, I also have this concern. One possible method is checking the module class whether there is 'quantize' string in its name.

**Review by PenghuiCheng — APPROVED** [review]

File: examples/text-classification/run_glue.py (hunk 1 above)
Comment (echarlaix) [reply examples/text-classification/run_glue.py:760]: We could create a separate function verifying the model's quantization by comparing what layer is being quantized with the configuration describing the quantization process, but it might be a little bit heavy

**Comment by hshen14** [comment]
@xin3he  Please rebase the code. We will need to support reloading INT8 models.

**Review by PenghuiCheng — APPROVED** [review]
